pkg/ebpf/events_derived.go

```diff
@@ -116,3 +122,68 @@ func deriveContainerRemoved(t *Tracee) deriveFn {
 		return trace.Event{}, false, nil
 	}
 }
+
+func deriveDetectHookedSyscall(t *Tracee) deriveFn {
+	return func(event trace.Event) (trace.Event, bool, error) {
+		syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+		if err != nil {
+			return trace.Event{}, false, fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+		}
+		hookedSyscallData := t.ParseHookedAddresses(syscallsAdresses)
+		de := event
+		de.EventID = int(DetectHookedSyscallsEventID)
+		de.EventName = "hooked_syscalls"
+		de.ReturnValue = 0
+		de.Args = []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Name: "hooked_syscalls", Type: "hookedSyscallData[]"}, Value: hookedSyscallData},
+		}
+		de.ArgsNum = 1
+		return de, true, nil
+	}
+}
+
+func (t *Tracee) ParseHookedAddresses(addresses []uint64) []bufferdecoder.HookedSyscallData {
+	hookedSyscallData := make([]bufferdecoder.HookedSyscallData, 0, 0)
+	for idx, syscallsAdress := range addresses {
+		InTextSegment, err := t.kernelSymbols.TextSegmentContains(syscallsAdress)
+		if err != nil {
+			continue
+		}
+		if !InTextSegment {
+			hookingFunction := t.ParseSymbol(syscallsAdress)
+			arch := t.config.OsConfig.GetOSReleaseFieldValue(helpers.OS_ARCH)
+			var syscallNumber int32
+			if strings.Compare(arch, "x86_64") == 0 {
+				if idx > len(syscallsToCheckX86) {
+					continue
+				}
+				syscallNumber = int32(syscallsToCheckX86[idx])
+			} else {
+				if idx > len(syscallsToCheckArm) {
+					continue
+				}
+				syscallNumber = int32(syscallsToCheckArm[idx])
+			}
+			event, found := EventsDefinitions[syscallNumber]
+			var hookedSyscall bufferdecoder.HookedSyscallData
+			if found {
+				hookedSyscall = bufferdecoder.HookedSyscallData{event.Name, hookingFunction.Owner}
+			} else {
+				hookedSyscall = bufferdecoder.HookedSyscallData{fmt.Sprint(syscallNumber), hookingFunction.Owner}
+			}
+			hookedSyscallData = append(hookedSyscallData, hookedSyscall)
+		}
+	}
+	return hookedSyscallData
+}
+
+func (t *Tracee) ParseSymbol(address uint64) *helpers.KernelSymbol {
```

Comment:
This function shouldn't be a member of tracee, it should only receive the `t.kernelSymbols` as argument.